pagopa/io-app
# chore: [IA-842] Integrate React Native Vision Camera to scan QR Codes

## Short description
This PR is going to replace the current QR Code scanner library with `react-native-vision-camera`. This is just a preparation to handle more barcodes in the scanning section. The scanning functionality should remain invariated since just the library under the hood has been replaced.

📜 [Here's a document](https://pagopa.atlassian.net/wiki/spaces/IOAPP/pages/488374611/Lettura+del+codice+Data+Matrix) to better understand the changes and the tradeoff

⚠️ This PR will increase the app's minimum iOS version from `10` to `11`.

<details>
<summary>iOS Preview</summary>

https://user-images.githubusercontent.com/5963683/166224728-b0aa9908-20b6-4177-902d-065a3f938ddc.mov

</details>

<details>
<summary>Android Preview</summary>

https://user-images.githubusercontent.com/5963683/166250718-c14e7a07-0e4f-48ac-a5c9-04151e9bd7a2.mp4

</details>

## List of changes proposed in this pull request
- Created a `BarcodeCamera` component to separate the specific library code from the rest of the codebase
- Integrated the new `BarcodeCamera` component replacing the old `react-native-qrcode-scanner` library

## How to test
Remove all the caches as described in [the document](https://pagopa.atlassian.net/wiki/spaces/IOAPP/pages/488374611/Lettura+del+codice+Data+Matrix) and do a fresh install of the application through `yarn run-ios` or `yarn run-android`. Going in the **Wallet** -> **Pay notice**, the QR Code scanner should work as expected.
